### PR TITLE
Codegen migration leftovers

### PR DESCRIPTION
**Pull request by aymannadeem**
https://github.com/github/semantic/pull/448 moved AST generation code from `tree-sitter` into `Semantic`. [Some tests introduced in `tree-sitter`](https://github.com/tree-sitter/haskell-tree-sitter/pull/243/files) have yet to be ported over, which is what I'm doing here so I can 🔥them in `tree-sitter`.

I am moving tests and updating cabal files with test-suites for:
- [x] Go
- [x] Java
- [x] JSON
- [x] Python
- [x] Ruby
- [x] TypeScript
- [x] TSX
- [x] PHP?

Other things:
- [x] I also need to form an equivalent of `tree-sitter-test-helpers` in `semantic`. I plan on moving it to `semantic-codegen` so we don't have to create a new package. 
- [x] we shouldn’t import `tree_sitter_json` from `Language.JSON.Grammar`, because then loading `Language.JSON` in the REPL (for example) requires loading `Language.JSON.Grammar`; instead, we should continue to get it from `TreeSitter.JSON`. Ditto for each of the other langs.
- [ ]  For `JSON` specifically `Language.JSON.Grammar` shouldn’t exist at all because we don’t do any à la carte anything with it. Same goes for `Java` and languages we don't have Assignment for.
- [x] Submodules are a _pain_. We currently bring in the entire submodule just for the `node_types.json` file. To better manage this without manually hard-copying it over and risking being out-of-sync with each update, @dcreager proposed we could have a data file in `haskell-tree-sitter` that gets installed with the package. We have `getDataFileName :: FilePath -> IO FilePath` defined in the autogenerated `Paths_` module. If we do something like `getNodeTypesPath :: IO FilePath`, we could have that function import `Paths_tree_sitter_LANG` and use `getDataFileName "relative/path/to/node-types/file"`. Then we can nuke the submodules in Semantic and use something like `runIO TypeScript.getNodeTypesPath >>= astDeclarationsForLanguage Grammar.tree_sitter_typescript`
- [x] rename `Marshal.JSON` to `AST.Marshal.JSON` 
- [x] @robrix has created a new `semantic-parse` package and moved that into the `semantic-ast` executable component into `semantic-parse`. I've merged in his branch and will merge `semantic-codegen` back into `semantic-ast`, which means `semantic-python` can depend on `semantic-ast`, `semantic-parse` can depend on `semantic-python`.

How does that sit with everyone? @github/semantic-code

### CODE REVIEW

**Comment by aymannadeem** [comment]
Couple notes:

- I've disabled PHP temporarily. I wasn't sure if it was okay to delete it all just yet since we'll eventually be replacing it anyway; @patrickt suggested a less nuclear option with just removing it from `semantic.cabal` and `phpParser` from `Parsing.Parser` but the Assignment file's still intact.
- I've added a `getNodeTypesPath` to eventually deprecate submodules, but right now we also rely on vendored dependencies for `addDependentFileRelative "../../../vendor/tree-sitter-php/src/parser.c"` for all languages, so we'll need to do something there in order for that not to break once submodules go away. 
- > For JSON specifically Language.JSON.Grammar shouldn’t exist at all because we don’t do any à la carte anything with it. Same goes for Java and languages we don't have Assignment for.

  Keeping these for now.

**Comment by tclem** [comment]
> * I've disabled PHP temporarily.

The tricky part here is that we do support it for ToC summaries (which requires assignment). There's been some talk of removing that feature to ease the transition, but I don't know if a decision has been made.

**Comment by patrickt** [comment]
@tclem We could, if necessary, fix up PHP assignment…

**Comment by aymannadeem** [comment]
@tclem @patrickt I guess it depends on what the work to fix assignment errors is vs. full migration away from it?

File: semantic-go/src/Language/Go/AST.hs [hunk 1]
```diff
@@ -16,6 +16,7 @@ module Language.Go.AST
 
 import           Prelude hiding (False, Float, Integer, Rational, String, True)
 import           AST.GenerateSyntax
-import qualified Language.Go.Grammar as Grammar
+import           Language.Haskell.TH.Syntax (runIO)
+import qualified TreeSitter.Go as Go (getNodeTypesPath, tree_sitter_go)
 
-astDeclarationsForLanguage Grammar.tree_sitter_go "../../../vendor/tree-sitter-go/src/node-types.json"
+runIO Go.getNodeTypesPath >>= astDeclarationsForLanguage Go.tree_sitter_go
```
Comment (patrickt) [inline semantic-go/src/Language/Go/AST.hs]: Loving it.

File: semantic-ast/src/AST/Test.hs [hunk 2]
```diff
@@ -0,0 +1,80 @@
+{-# LANGUAGE OverloadedStrings #-}
+module AST.Test
```
[the remark this answers is not in the document]
Comment (aymannadeem) [reply semantic-ast/src/AST/Test.hs]: word, that's misleading.

**Comment by aymannadeem** [comment]
After talking with @rewinfrey in Slack, it seems that since PHP is a language we support for Code Nav and not just ToC, we should pause and not temporarily disable this just yet. It would feel safer to be more intentional and do this while having a transition plan to shift toward precise that doesn't break code nav.

**Pull request merged**